From d1423d828d0a063a7928931c98aedf29cc165235 Mon Sep 17 00:00:00 2001
From: Gerhard Baer <Gerhard.Baer@protonmail.com>
Date: Tue, 29 Oct 2024 12:14:09 +0100
Subject: [PATCH] cmake: fix build with StreamPU shared-lib only.

Try to link against StreamPU shared libs when using system-wide
installation without static libs.

Signed-off-by: Gerhard Baer <Gerhard.Baer@protonmail.com>
---
 CMakeLists.txt | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cca36b83e..8c8625226 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -449,8 +449,13 @@ endif()
 if (NOT AFF3CT_INCLUDE_SPU_LIB)
     find_package(StreamPU CONFIG 1.0.2 REQUIRED)
     message(STATUS "AFF3CT - Library found: StreamPU")
-    target_link_libraries(aff3ct-obj PUBLIC spu::spu-static-lib)
-    aff3ct_target_link_libraries(PUBLIC spu::spu-static-lib)
+    if (AFF3CT_COMPILE_SHARED_LIB)
+        target_link_libraries(aff3ct-obj PUBLIC spu::spu-shared-lib)
+        aff3ct_target_link_libraries(PUBLIC spu::spu-shared-lib)
+    else()
+        target_link_libraries(aff3ct-obj PUBLIC spu::spu-static-lib)
+        aff3ct_target_link_libraries(PUBLIC spu::spu-static-lib)
+    endif()
 endif()
 
 # GSL
-- 
2.39.5

